### PR TITLE
PoC Symfony workflow

### PR DESCRIPTION
**Pull request by loic425**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.13                  |
| Bug fix?        | no                                                       |
| New feature?    | yes                                                       |
| BC breaks?      | no                                                       |
| Deprecations?   | no |
| Related tickets |      |
| License         | MIT                                                          |

### Catalog promotion

- [x] State machine configuration

### Order

- [x] State machine configuration

**Before Placed Order**
- [x] sylius_assign_number
- [x] sylius_assign_token

**After Placed Order**
- [x] sylius_request_shipping
- [x] sylius_request_payment
- [x] sylius_create_payment
- [x] sylius_create_shipment
- [x] sylius_hold_inventory
- [x] sylius_increment_promotions_usages
- [x] sylius_save_addresses_on_customer
- [x] sylius_set_order_immutable_names

**After Canceled Order**
- [x] sylius_cancel_payment
- [x] sylius_cancel_shipment
- [x] sylius_cancel_order_payment
- [x] sylius_cancel_order_shipment
- [x] sylius_cancel_order_inventory
- [x] sylius_decrement_promotions_usages

### Order checkout

- [x] State machine configuration

**After Completed Checkout**
- [x] sylius_create_order
- [x] sylius_save_checkout_completion_date
- [x] sylius_control_payment_state
- [x] sylius_control_shipping_state

**After Selected Shipping**
- [x] sylius_process_cart
- [x] sylius_skip_payment

**After Addressed**
- [x] sylius_process_cart

**After Selected Payment**
- [x] sylius_process_cart

**After Skipped Shipping**
- [x] sylius_process_cart

#### After Skipped Payment
- [x] sylius_process_cart

### Order payment

- [x] State machine configuration

**After Paid**
- [x] sylius_order_paid
- [x] sylius_resolve_state

### Order shipping

- [x] State machine configuration

**After Shipped**
- [x] sylius_resolve_state

### Payment

- [x] State machine configuration

**After Failed Payment**
- [x] sylius_process_order

**After Canceled Payment**
- [x] sylius_process_order

**After Completed Payment**
- [x] sylius_resolve_state

**After Processed Payment**
- [x] sylius_resolve_state

**After Refunded Payment**
- [x] sylius_resolve_state

**After Authorized Payment**
- [x] sylius_resolve_state

### Product Review

- [x] State machine configuration

**After Accepted Product Review**
- [x] sylius_update_rating

### Shipment

- [x] State machine configuration

**Before Shipped**
- [x] sylius_assign_date

**After Shipped**
- [x] sylius_resolve_state

### CODE REVIEW

File: config/packages/_sylius.yaml [hunk 1]
```diff
@@ -16,3 +16,7 @@ sylius_api:
 sylius_shop:
     product_grid:
         include_all_descendants: true
+
+sylius_resource:
+    settings:
+        state_machine_component: symfony
```
Comment (loic425) [inline config/packages/_sylius.yaml:22]: To remove of course, Developer has to choose himself/herself

Comment (vasilvestre) [reply config/packages/_sylius.yaml:22]: Wouldn't it be ok to make it the new default value tho?
The upgrade guide needs to get a new part about this BC but new projects should use symfony/workflow, shouldn't they?

Comment (vasilvestre) [reply config/packages/_sylius.yaml:22]: Wouldn't it be ok to make it the new default value tho?
The upgrade guide needs to get a new part about this BC but new projects should use symfony/workflow, shouldn't they?

Comment (loic425) [reply config/packages/_sylius.yaml:22]: Hum, by default, the state machine component is set up to winzou when both are available.
This configuration could stay like this cause it does not impact end-user application.

But still, it should stay with Winzou for 1.x release at least for 1.13.

Comment (loic425) [reply config/packages/_sylius.yaml:22]: But in Sylius standard we'll be able to set this value to use Symfony Workflow

Comment (vasilvestre) [reply config/packages/_sylius.yaml:22]: How to ensure both workflow system are maintained ? We can't run all test twice

Comment (loic425) [reply config/packages/_sylius.yaml:22]: We can "include" a test with Workflow it's easy to do.

File: src/Sylius/Bundle/CoreBundle/Resources/config/services/workflow.xml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+ (c) Paweł Jędrzejewski
+
+ For the full copyright and license information, please view the LICENSE
+ file that was distributed with this source code.
+
+-->
+
+<container xmlns="http://symfony.com/schema/dic/services" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+    <imports>
```
Comment (diimpp) [inline src/Sylius/Bundle/CoreBundle/Resources/config/services/workflow.xml:15]:
```patch
+++ <imports>
+++     <import resource="workflows/*.xml" />
+++     <import resource="workflows/listeners/*.xml" />
+++     <import resource="workflows/callbacks/*.xml" />
+++ </imports>
```
I would vote for wildcards like so.

In long term maintainability, chance to forget to import file or remove import without removing file becomes high, happened to me multiple times, before I've switched to wildcards like in the example.

**Comment by jakubtobiasz** [comment]
Closing as we implemented the support for Symfony Workflow. Thanks for your PR that inspired us a lot 🎉!

**Pull request closed without merging**